pkg/api/operators/apis/suggestion/v1alpha2/suggestion_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SuggestionSpec defines the desired state of Suggestion
+type SuggestionSpec struct {
+	// Number of desired pods. This is a pointer to distinguish between explicit
+	// zero and not specified. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Type of the suggestion.
+	Type SuggestionType `json:"type,omitempty"`
+
+	// Defines if the suggestion needs previous results.
+	NeedHistory bool `json:"needHistory,omitempty"`
```

Comment:
If NeedHistory == true, what does it mean for ES and HP? For HP, it means it needs trial final metrics (opposite: random search), while in ES, it needs periodical metrics?